### PR TITLE
OCPBUGS-1411: Bump k8s dependencies to 1.25

### PR DESCRIPTION
**Pull request by lobziik**

### CODE REVIEW

**Review by damdo — COMMENTED** [review]
Provided tests pass:
/lgtm

**Review by damdo — COMMENTED** [review]
/lgtm

File: .golangci.yaml [hunk 1]
```diff
@@ -4,16 +4,13 @@ linters:
   # https://golangci-lint.run/usage/linters/#enabled-by-default-linters
   enable:
     # Default linters
-    - deadcode
```
Comment (JoelSpeed) [inline .golangci.yaml:7]: Why are we dropping linters?

**Review by JoelSpeed — COMMENTED** [review]
/approve
/lgtm

File: .golangci.yaml (hunk 1 above)
Comment (lobziik) [reply .golangci.yaml:7]: It was deprecated and might be covered with the `unused` one.
Some context around: https://github.com/golangci/golangci-lint/pull/3125

**Pull request merged**